### PR TITLE
BI-4109 create swagger spec

### PR DESCRIPTION
**Pull request by markpit**
Ditched the idea for multiple reasons. As we have to use HATEOS links, we would need new reason endpoints to get and add reasons.
Not sure if Patch should return the resource - at the moment it is.
Some http status' might not be correct

### CODE REVIEW

File: spec/schema.json [hunk 1]
```diff
@@ -16,53 +16,346 @@
...
+        "summary": "Create a new strike-off objection request",
         "produces": [
           "application/json"
         ],
+        "tags": [
+          "Strike Off Objections"
+        ],
+        "responses": {
+          "201": {
+            "description": "An objection request has been received.",
```
Comment (mwestacott) [inline spec/schema.json:52]: Objection request has been 'created' (as well as 'received').

File: spec/schema.json [hunk 2]
```diff
@@ -16,53 +16,346 @@
...
+            "schema": {
+              "$ref": "#/definitions/ResourceCreated"
```
Comment (mwestacott) [inline spec/schema.json:54]: Could be more specific here - 'ObjectionCreated' maybe.

Comment (markpit) [reply spec/schema.json:54]: the same object structure is returned by post attachment as well so I named it ResourceCreated for re-use. I could name it ObjectionCreated but I'd also have to create an AttachmentCreated as well.

File: spec/schema.json (hunk 1 above)
Comment (markpit) [reply spec/schema.json:52]: ok

File: spec/schema.json (hunk 2 above)
Comment (mwestacott) [reply spec/schema.json:54]: Ok. Fair enough. Confluence page had different field names being returned (objectionId and fileId) but I'm happy with the generic approach.

File: spec/schema.json [hunk 3]
```diff
@@ -16,53 +16,346 @@
...
+        {
+          "name": "objectionId",
+          "in": "path",
+          "required": true,
+          "description": "ID of the Objection",
+          "type": "string"
+        }
+      ],
+      "get": {
+        "summary": "Returns the objection request identified by objectionId",
```
Comment (mwestacott) [inline spec/schema.json:71]: Not an 'objection request' here, just an 'objection'.

File: spec/schema.json [hunk 4]
```diff
@@ -16,53 +16,346 @@
...
+          "404": {
+            "description": "not found"
+          }
+        }
+      }
+    },
+    "/": {
       "post": {
-        "summary": "Temporary end-point to prove sprint zero connectivity from the web app",
+        "summary": "Create a new strike-off objection request",
```
Comment (mwestacott) [inline spec/schema.json:43]: I think 'request' should go here. In a few other places too. HTTP requests are coming in, as you pointed out, but our terminology now doesn't have 'objection requests'.

File: spec/schema.json [hunk 5]
```diff
@@ -16,53 +16,346 @@
...
+        "summary": "Returns the objection request identified by objectionId",
+        "produces": [
+          "application/json"
+        ],
+        "tags": [
+          "Strike Off Objections"
+        ],
+        "responses": {
+          "200": {
+            "description": "a Status object has been returned successfully",
```
Comment (mwestacott) [inline spec/schema.json:80]: 'An objection has been returned successfully'.

File: spec/schema.json [hunk 6]
```diff
@@ -16,53 +16,346 @@
...
             "schema": {
-              "$ref": "#/definitions/ObjectionOutput"
+              "$ref": "#/definitions/ResourceCreated"
             }
+          },
+          "400": {
+            "description": "there was an error processing the request"
+          },
+          "404": {
+            "description": "the request or reason does not exist"
```
Comment (mwestacott) [inline spec/schema.json:140]: I think this should be 'the objection does not exist'.

File: spec/schema.json [hunk 7]
```diff
@@ -16,53 +16,346 @@
...
+        }
+      }
+    },
+    "CreatedBy": {
+      "type": "object",
+      "properties": {
+        "id": {
+          "type": "string"
+        },
+        "forename": {
```
Comment (mwestacott) [inline spec/schema.json:330]: Where / how will we get forename and surname from? I thought that user just signs in with eMail address.

File: spec/schema.json (hunk 3 above)
Comment (markpit) [reply spec/schema.json]: updated

File: spec/schema.json (hunk 4 above)
Comment (markpit) [reply spec/schema.json]: updated

File: spec/schema.json [hunk 8]
```diff
@@ -16,53 +16,346 @@
...
+        },
+        "status": {
+          "$ref": "#/definitions/Status"
         }
       }
+    },
+    "Status": {
+      "type": "string",
+      "enum": [
+        "open",
```
Comment (mwestacott) [inline spec/schema.json:356]: Personally I like enum values in uppercase. I just think it helps them stand out (a bit like a constant) but up to you.

File: spec/schema.json (hunk 5 above)
Comment (markpit) [reply spec/schema.json]: updated

File: spec/schema.json (hunk 6 above)
Comment (markpit) [reply spec/schema.json]: updated

File: spec/schema.json (hunk 7 above)
Comment (markpit) [reply spec/schema.json:330]: we get them from ERIC, I guess we don't technically need them as we can cross reference the user data

Comment (mwestacott) [reply spec/schema.json:330]: Fair enough if we have them.

Comment (markpit) [reply spec/schema.json]: i've taken them out, they were just nice to have's

File: spec/schema.json (hunk 8 above)
Comment (markpit) [reply spec/schema.json]: updated

**Pull request merged**